src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const item of action.keysToRemove) {
+          delete state[item];
+        }
+
+        break;
+
+      case 'clear':
+        for (const item in state) {
+          delete state[item];
+        }
+
+        break;
+
+      default:
+        throw new Error('Error');
```

Comment:
It will be better to submit a bug with a minimal description of the bug.

For example:
`'Undefined action type'`